### ISSUE TITLE
Create a `Photon` class that can interact with `Particle` instances

### ISSUE THREAD

**Issue by namurphy** [PlasmaPy/PlasmaPy#1175]
Another question that came up during the SULI summer school: is there a class for photons?  There currently is not, though photons often interact with particles and so it is something that we should include.

**Comment by namurphy** [comment]
One thought in relation to #1174...we should have an error for operations like
```Python
positron + electron > Photon()
```
This is because two photons need to be released for conservation of momentum.

**Comment by namurphy** [comment]
Here are some of my thoughts on what the `Photon` class should entail.  
 - `Photon` should most likely be subclassed from `plasmapy.particles.particle_class.AbstractParticle`.  
 - There should be one input parameter, which could have units of energy (e.g., `u.J`), wavelength (`u.m`), frequency (`u.s ** -1`), or momentum (`u.kg * u.m / u.s`)
 - We can use `validate_quantities` from `plasmapy.utils.decorators.checks` (I think)
 - The are a few attributes that should be hard-coded for a `Photon` instance (where `u` ≡ `astropy.units`).
    - `photon.mass` → `0 * u.kg`
    - `photon.charge` → `0 * u.C`
    - `photon.charge_number` → `0`
    - `photon.spin` → `1`
    - `photon.half_life` → `np.inf * u.s`
  - There are probably a few more attributes that are included in `Particle` that could also be included in `Photon`
  - There should be a few attributes that will be defined from the : 
    - `photon.energy`: E = hν
    - `photon.frequency`: ν
    - `photon.wavelength`: λ
    - `photon.momentum`: p
    If no arguments were provided, then each of these things should be `numpy.nan` in the appropriate units.

I'll try to add more later.  Thank you for working on this!

**Comment by RAJAGOPALAN-GANGADHARAN** [comment]
Thanks for the info @namurphy , working on this.

**Comment by RAJAGOPALAN-GANGADHARAN** [comment]
Hello and Sorry for the delay, can I still work on this? I sorta have small change. I have created this class under `particle_class`, is this place fine or it needs to be in separate file under particles say photon_class?

**Comment by namurphy** [comment]
Yes, please feel free to work on this!  I think `particle_class.py` is a good place to put it for now.   

We will want to enable `Photon` to accept a `Quantity` with units of energy, wavelength, or frequency.  The good news is that we can use an [equivalency](https://docs.astropy.org/en/stable/units/equivalencies.html) from Astropy ([`spectral`](https://docs.astropy.org/en/stable/units/equivalencies.html#spectral-units)) to do these conversions for us. Also, energy, wavelength, and frequency will all be positive.

Thank you again!

**Comment by RAJAGOPALAN-GANGADHARAN** [comment]
How do we account for the wave nature of a Photon if Photon is abstracted from Particle. From my physics class I remember photon acts like both particle and wave. I'm wondering how wave function and probabilities say using Schrodinger's equation can be calculated for a Photon with current implementation. Pardon me I'm a rookie in Physics :P
